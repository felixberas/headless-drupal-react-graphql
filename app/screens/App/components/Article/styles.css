.Article {
  margin: 10px 0;
}

.Header {
  font-size: 20px;
  font-weight: bold;
}

.Body {
  font-size: 14px;
}
