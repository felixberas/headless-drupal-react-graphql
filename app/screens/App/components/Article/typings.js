// @flow

export type ArticleProps = {
  id: string,
  title: string,
  body: ?string,
  articleColor: string,
  setRandomColor: Function,
};
