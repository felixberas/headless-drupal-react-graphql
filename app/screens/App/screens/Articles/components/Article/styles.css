.Wrapper {
  margin: auto;
  max-width: 500px;
  margin-bottom: 2rem;
}

.Title {
  font-size: 1.1rem;
}

.Body {
  margin-top: 1rem;
}
