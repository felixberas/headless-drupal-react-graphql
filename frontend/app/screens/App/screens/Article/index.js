// @flow

import React from 'react';
import Helmet from 'react-helmet';
import gql from 'graphql-tag';

type ArticleProps = {
  title: string,
};

const Article = ({
  title,
}: ArticleProps): React.Element<any> => (
  <div>
    <Helmet title={title} />
    <h1>{title}</h1>
  </div>
);

Article.fragments = {
  articleFragment: gql`
    fragment ArticleFragment on Article {
      title
    }
  `,
};

export default Article;
