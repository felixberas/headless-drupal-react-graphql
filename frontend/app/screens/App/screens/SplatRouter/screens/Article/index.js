// @flow

import React from 'react';
import Helmet from 'react-helmet';
import { Link } from 'react-router';
import gql from 'graphql-tag';
import styled from 'styled-components';

type ArticleProps = {
  title: string,
  body: string,
};

const Wrapper = styled.div`
  background-color: Red;
  margin: 1rem 0;
`;

const Title = styled.h1`
  font-weight: bold;
  font-size: 24px;
  margin-bottom: 1rem;
`;

const Body = styled.div`
  font-size: 16px;
`;

const Article = ({ title, body }: ArticleProps): React.Element<any> => (
  <Wrapper>
    <Helmet title={title} />
    <Title>{title}</Title>
    {/* eslint-disable react/no-danger */}
    <Body dangerouslySetInnerHTML={{ __html: body }} />
    {/* eslint-enable react/no-danger */}
    <Link to="/articles">Back to overview</Link>
  </Wrapper>
);

Article.fragments = {
  articleFragment: gql`
    fragment ArticleFragment on NodeArticle {
      title
      body
    }
  `,
};

export default Article;
