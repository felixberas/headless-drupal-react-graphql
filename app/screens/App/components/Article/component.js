// @flow

import React from 'react';
import type { ArticleProps } from './typings';
import styles from './styles.css';

const Article = ({ title, body, setRandomColor, articleColor } : ArticleProps) => (
  <div className={styles.Article} style={{ backgroundColor: articleColor }}>
    <button onClick={setRandomColor}>Color magic</button>
    <h2 className={styles.Header}>{title}</h2>
    <p className={styles.Body}>{body}</p>
  </div>
);

export default Article;
