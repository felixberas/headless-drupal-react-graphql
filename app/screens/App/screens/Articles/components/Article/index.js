import React from 'react';
import Relay, { createContainer } from 'react-relay';
import styles from './styles.css';

const Article = ({ article }) =>
  <div className={styles.Wrapper}>
    <h2 className={styles.Title}>{article.title}</h2>
    <p className={styles.Body}>{article.body}</p>
  </div>;

export default createContainer(Article, {
  fragments: {
    article: () => Relay.QL`
      fragment on Article {
        title
        body
      }
    `,
  },
});
