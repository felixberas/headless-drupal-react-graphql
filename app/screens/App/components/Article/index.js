import compose from 'recompose/compose';
import withState from 'recompose/withState';
import mapProps from 'recompose/mapProps';
import Article from './component';

const randomColor = () => {
  const letters = '0123456789ABCDEF';
  
  return Array
    .from(Array(6))
    .reduce((color) => color + letters[Math.floor(Math.random() * 16)], '#');
};

const withColorState = withState('articleColor', 'setArticleColor', 'inherit');
const withRandomColorSetter = mapProps(({ setArticleColor, ...props }) => ({
  ...props,
  setRandomColor : () => setArticleColor(randomColor()),
}));

export default compose(
  withColorState,
  withRandomColorSetter,
)(Article);
