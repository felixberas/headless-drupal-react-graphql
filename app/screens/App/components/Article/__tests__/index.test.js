import React from 'react';
import Article from 'Article';
import { shallow } from 'enzyme';

describe('article component', () => {
  it('renders as expected', () => {
    const component = shallow(
      <Article title="Lorem ipsum" body="Dolor sit amet" id="1" />
    );

    expect(component.html()).toMatchSnapshot();
  });

  it('changes color on click', () => {
    const component = shallow(
      <Article title="Lorem ipsum" body="Dolor sit amet" id="1" />
    );

    expect(component).to.have.style('background-color', 'inherit');

    const button = component.find('button');
    button.simulate('click');

    expect(component).to.not.have.style('background-color', 'inherit');
  });
});
