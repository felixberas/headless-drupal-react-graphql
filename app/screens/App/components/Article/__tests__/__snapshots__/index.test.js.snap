exports[`article component renders as expected 1`] = `"<div class=\"Article\" style=\"background-color:inherit;\"><button>Color magic</button><h2 class=\"Header\">Lorem ipsum</h2><p class=\"Body\">Dolor sit amet</p></div>"`;
